targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF51/sleep.c

```diff
@@ -64,9 +65,10 @@ void hal_sleep(void)
         __SEV();
         __WFE();
 
-        // Test if there is an interrupt pending (mask reserved regions)
-        if (SCB->ICSR & (SCB_ICSR_RESERVED_BITS_MASK)) {
-            // Ok, there is an interrut pending, no need to go to sleep
+        // Test if we should enter WFE
+        if (SCB->ICSR & (SCB_ICSR_WFE_MASK)) {
```

Comment:
Is the ICSR check needed? If an interrupt is pending shouldn't `__WFE()` just clear the event register and return immediately?

Reply:
I'm not confident enough to say that we don't need the ICSR check but I think it's prudent.  At the minimum it prevents entry to sleep while in an exception handler (either accidentally or intentionally) by including VECTACTIVE in the mask.